Mathlib/Data/Matrix/Basis.lean

```diff
@@ -70,14 +70,12 @@ theorem mulVec_stdBasisMatrix [NonUnitalNonAssocSemiring α] [Fintype m]
 
 theorem matrix_eq_sum_stdBasisMatrix [AddCommMonoid α] [Fintype m] [Fintype n] (x : Matrix m n α) :
     x = ∑ i : m, ∑ j : n, stdBasisMatrix i j (x i j) := by
-  ext i j; symm
-  iterate 2 rw [Finset.sum_apply]
-  convert (Fintype.sum_eq_single i ?_).trans ?_; swap
-  · -- Porting note(#12717): `simp` seems unwilling to apply `Fintype.sum_apply`
-    simp (config := { unfoldPartialApp := true }) [stdBasisMatrix, (Fintype.sum_apply)]
-  · intro j' hj'
-    -- Porting note(#12717): `simp` seems unwilling to apply `Fintype.sum_apply`
-    simp (config := { unfoldPartialApp := true }) [stdBasisMatrix, (Fintype.sum_apply), hj']
+  ext i j
+  rw [← Fintype.sum_prod_type']
+  -- In this proof, we abuse the definition of `Matrix` by applying matrices.
```

Comment:
This is not abuse of `Matrix`, this is how it is intended to be used. Of course, it might be abuse of Lean itself... (vs using a one-field structure).

I think the motivation applies to the existence of `Matrix.add_apply` etc too; but it's interesting that somehow the `(some_lemma)` syntax makes things work.

Reply:
What do you want me to write, concretely?

Reply:
I think what I actually want you to write `Matrix.sum_apply`, and use that lemma here instead of `Fintype.sum_apply`.

Reply:
I was hoping for a better fix, but okay

Reply:
> I was hoping for a better fix, but okay

You could extend the comments around `Matrix.add_apply` etc if you want to push for that.